src/main/java/com/iri/movietickets/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package com.iri.movietickets.security;
+
+import com.iri.movietickets.exception.AuthenticationException;
+import com.iri.movietickets.lib.Inject;
+import com.iri.movietickets.lib.Service;
+import com.iri.movietickets.model.User;
+import com.iri.movietickets.service.UserService;
+import com.iri.movietickets.util.HashUtil;
+import java.util.Optional;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> user = userService.findByEmail(email);
+        if (user.isPresent()) {
+            String hashedPassword = HashUtil.hashPassword(password, user.get().getSalt());
+            if (user.get().getPassword().equals(hashedPassword)) {
+                return user.get();
+            } else {
+                throw new AuthenticationException("Wrong email or password");
+            }
+        } else {
+            throw new NullPointerException("This user has not registered yet");
+        }
```

Comment:
```suggestion
                return user.get();
            } 
        }
        throw new AuthenticationException("Wrong email or password");
```
do we need else after `return`?
also, let's throw an exception only once if any of the conditions are not met

Reply:
Теперь окончательно запутался
Если пароль совпадает - то возвращаем юзера если нет то эксепшин 
Ну и если нет такого обьекта нпе 
Необходимо пояснение

Reply:
в обоих случаях - и если логин неправильный (то есть если такого юзера нет в базе), и если пароль неправильный - нам лучше кидать один ексепшн с универсальным сообщением "неправильный логин или пароль"

чтобы если кто-то пытается взломать аккаунт, то мы не наводили его на решение проблемы (не подошел конкретно логин или конкретно пароль)